Fix/go mod versioning

repjarms opened the pull request:
### Summary

This PR fixes an issue with the `go.mod` file where we had not specified the `/v2` path segment in the `go.mod`. This caused incompatibilities with installing v2 of the SDK, and caused it to revert to `1.7.0` when calling `go get`.

This PR will release version `2.0.2` of the SDK.

mediumTaj reviewed (approved):
👍 looks good!

padamstx commented on .travis.yml:
```diff
@@ -28,10 +28,12 @@ before_script:
 
 script:
 # This step is required since we use a go module
+- cd ./v2
```
Because working with the project is a little more tedious with the top level version directory present, I actually added a Makefile to the platform-services-go-sdk project.   Check it out to see if it might be useful for you as well:
https://github.com/IBM/platform-services-go-sdk/blob/master/Makefile
https://github.com/IBM/platform-services-go-sdk/blob/master/.travis.yml

With that sort of setup, I can run the following commands at the project root:
```
make all
make build
make unittest
make tidy
make lint
etc.
```

repjarms replied:
@padamstx that's a good point. I will likely add Make and update the configuration in a future PR so that we can prioritize this release to resolve issues with moving to the latest major.

padamstx commented on v2/go.sum:
```diff
@@ -1,6 +1,6 @@
 github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
-github.com/IBM/go-sdk-core/v4 v4.8.1 h1:gFcQcQGOoHg+8pNILVTgGMh77SG756XkOatv/gs2VTg=
-github.com/IBM/go-sdk-core/v4 v4.8.1/go.mod h1:GECJ/p0r9Hs0XcOnCiGjZrt6M/rQc+gW8YkCsGJ+j5U=
+github.com/IBM/go-sdk-core v0.0.0-20200217212347-fe152a0e9e89 h1:xvNuMMhsiGhHUA7DWob/sM8OFddqDdzOJkpX626cA1I=
```
this doesn't look right to me.  we shouldn't have any code in your project that imports the go core without the "v4" part being included in the import path.

padamstx commented on v2/go.sum:
```diff
@@ -1,6 +1,6 @@
-github.com/IBM/go-sdk-core/v4 v4.8.1/go.mod h1:GECJ/p0r9Hs0XcOnCiGjZrt6M/rQc+gW8YkCsGJ+j5U=
+github.com/IBM/go-sdk-core v0.0.0-20200217212347-fe152a0e9e89 h1:xvNuMMhsiGhHUA7DWob/sM8OFddqDdzOJkpX626cA1I=
+github.com/IBM/go-sdk-core v0.0.0-20200217212347-fe152a0e9e89/go.mod h1:2pcx9YWsIsZ3I7kH+1amiAkXvLTZtAq9kbxsfXilSoY=
 github.com/IBM/go-sdk-core/v4 v4.9.0 h1:OkSg5kaEfVoNuBA4IsIOz8Ur5rbGHbWxmWCZ7nK/oc0=
```
should probably be using the 4.9.4 release of the go core (the latest)

padamstx commented on v2/go.mod:
```diff
@@ -10,5 +10,5 @@ require (
 	github.com/onsi/gomega v1.10.3
 	github.com/stretchr/testify v1.6.1
-	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
+	github.com/watson-developer-cloud/go-sdk v1.7.0
```
you shouldn't need a "require" entry for your own library.  do you still have some code that uses the OLD 1.7.0 version of the go sdk???

padamstx commented on v2/go.sum:
```diff
@@ -18,13 +18,13 @@ github.com/fsnotify/fsnotify v1.4.9 h1:hsms1Qyu0jgnwNXIxa+/V/PDsU6CfLf6CNO8H7IWo
-github.com/go-openapi/strfmt v0.19.5/go.mod h1:eftuHTlB/dI8Uq8JJOyRlieZf+WkkxUuk0dgdHXr2Qk=
-github.com/go-openapi/strfmt v0.19.8 h1:9wAdSoImc5UCnUj79GhcjkJXxuU/nEwlbe6SKe9ZdRs=
-github.com/go-openapi/strfmt v0.19.8/go.mod h1:qBBipho+3EoIqn6YDI+4RnQEtj6jT/IdKm+PAlXxSUc=
+github.com/go-openapi/strfmt v0.19.3/go.mod h1:0yX7dbo8mKIvc3XSKp7MNfxw4JytCfCD6+bY1AVL9LU=
```
not sure why you have multiple versions of strfmt listed in go.sum.   the ps go-sdk only has this:
```
github.com/go-openapi/strfmt v0.19.10 h1:FEv6Pt/V4wLwP4vOCZbWlpfmi8kj4UiRip34IDE6SGw=
github.com/go-openapi/strfmt v0.19.10/go.mod h1:qBBipho+3EoIqn6YDI+4RnQEtj6jT/IdKm+PAlXxSUc=
```
and the go.mod has an explicit dependency on that package:
```
module github.com/IBM/platform-services-go-sdk

go 1.12

require (
	github.com/IBM/go-sdk-core/v4 v4.9.0
	github.com/go-openapi/strfmt v0.19.10
	github.com/google/uuid v1.1.1
	github.com/joho/godotenv v1.3.0
	github.com/onsi/ginkgo v1.14.2
	github.com/onsi/gomega v1.10.3
	github.com/satori/go.uuid v1.2.0
	github.com/stretchr/testify v1.6.1
)
```

padamstx reviewed (approved):
I left a few comments about dependencies that should probably be resolved before merging and along with a suggestion to add a Makefile to simplify various build/test steps, but otherwise the changes look good.

repjarms commented:
@padamstx I believe I've addressed the issues by regenerating the `go.sum`. I mentioned in the Make comment, but that is something I can look into on a future PR since it doesn't immediately block the release for now.

repjarms commented:
@padamstx @mediumTaj I updated the PR and did end up including the `Makefile` since that cleaned up a lot of the build process. Also I realized that the reason that the SDK was attempting to include itself as a module was due to the import statements in the generated code not specifying `v2` for the `common` package, which is provided within the SDK. Therefore, the go module resolver was attempting to find the `v1` package, and not finding it locally, retrieved it from the `1.7.0` version of the SDK.
